### PR TITLE
Add dependency injector

### PR DESCRIPTION
**Pull request by adrianclay**

### CODE REVIEW

File: components/DependencyInjector/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import { render, mount } from 'enzyme'
+import DependecyInjection from '../'
+
+describe('DependencyInjection', () => {
+  it('exposes rand', () => {
+    let m = mount(<DependecyInjection><div /></DependecyInjection>)
+    expect(m.instance().getChildContext().rand).toBe(Math.random)
```
Comment (craigjbass) [inline components/DependencyInjector/__tests__/index.js:8]: Is this defo asserting on the `<div>`

File: components/DependencyInjector/index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+export default class DependencyInjector extends React.Component {
```
Comment (craigjbass) [inline components/DependencyInjector/index.js:4]: I'd defo want to see a more generic DependencyInjector.

File: components/DependencyInjector/__tests__/index.js (hunk 1 above)
Comment (rrpff) [reply components/DependencyInjector/__tests__/index.js:8]: AFAICS the div here is doing nothing?

File: components/DependencyInjector/index.js (hunk 2 above)
Comment (rrpff) [reply components/DependencyInjector/index.js:4]: yeah this is a great proof of concept, but would be good to see an example we could use widely.

**Pull request closed without merging**